common/src/main/java/org/hiero/mirror/common/domain/transaction/HookExecutionCollector.java

```diff
@@ -0,0 +1,89 @@
+// SPDX-License-Identifier: Apache-2.0
+
+package org.hiero.mirror.common.domain.transaction;
+
+import java.util.ArrayDeque;
+import java.util.ArrayList;
+import java.util.List;
+import org.hiero.mirror.common.domain.hook.AbstractHook;
+
+/**
+ * Collects hook IDs for different execution phases and provides methods to build the final execution queue.
+ *
+ * This record encapsulates the three types of hook executions as specified in HIP-1195:
+ * 1. allowExecHookIds - PreTx hooks (HBAR, Token, NFT transfers)
+ * 2. allowPreExecHookIds - Pre hooks from PrePostTx (HBAR, Token, NFT transfers)
+ * 3. allowPostExecHookIds - Post hooks from PrePostTx (HBAR, Token, NFT transfers)
+ *
+ * The execution order is: allowExecHookIds → allowPreExecHookIds → allowPostExecHookIds
+ */
+public record HookExecutionCollector(
+        List<AbstractHook.Id> allowExecHookIds,
+        List<AbstractHook.Id> allowPreExecHookIds,
+        List<AbstractHook.Id> allowPostExecHookIds) {
+
+    /**
+     * Creates a new HookExecutionCollector with empty lists.
+     */
+    public static HookExecutionCollector create() {
+        return new HookExecutionCollector(new ArrayList<>(), new ArrayList<>(), new ArrayList<>());
+    }
+
+    /**
+     * Adds a hook ID to the allowExecHookIds list (PreTx hooks).
+     *
+     * @param hookId the hook ID to add
+     * @param ownerId the owner ID for the hook
+     * @return this collector for method chaining
+     */
+    public HookExecutionCollector addAllowExecHook(long hookId, long ownerId) {
```

Comment:
can do
```suggestion
    public HookExecutionCollector addAllowExecHook(final HookCall hookCall, long ownerId) {
        if (!hookCall.hasHookId()) {
            return;
        }
        
        // add. hook context
```

and the same to `addPrePostExecHook` to remove all `hookCall.hasHookId()` calls in `CryptoTransferTransactionHandler`.